core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2848,9 +2849,42 @@ private void renameInternal(RpcContext rpcContext, LockedInodePath srcInodePath,
 
     // Make sure destination path does not exist
     if (dstInodePath.fullPathExists()) {
-      throw new FileAlreadyExistsException(String
-          .format("Cannot rename because destination already exists. src: %s dst: %s",
-              srcInodePath.getUri(), dstInodePath.getUri()));
+      boolean failOnExist = true;
+      if (context.getOptions().hasS3SyntaxOptions()) {
```

Comment:
It's too long here, would you mind extracting it into a new method?

Reply:
done